### PR TITLE
Test -r flag

### PR DESCRIPTION
**Pull request by leotrs**
## Motivation
I broke this flag in the course of writing a different PR, so let's add this test!

## Notes
@ManimCommunity/tests should I be marking this test with `@pytest.mark.slow`? Not sure when to use this.

## Testing Status
Pass locally.

## Acknowledgement
- [x] I have read the [Contributing Guidelines](https://github.com/ManimCommunity/manim/wiki/Documentation-guidelines-(WIP))

### CODE REVIEW

**Review by huguesdevimeux — APPROVED** [review]
SGTM. Concerning the skip_slow flag, I think if the test takes more than 2 seconds to run it's safe to call it slow. 

We should have an empirical rule to determine whether a test is "slow"

**Comment by leotrs** [comment]
I mean I feel like whatever calls manim as a subcommand should be marked as slow.

**Comment by leotrs** [comment]
@naveen521kk I did `poetry add imageio` to add that library. Could you please let me know if I did this correctly? I don't know how to read/interpret the poetry files :sweat_smile: 

Also: I'm not feeling good about adding imageio as a dependency just for this one test.

**Comment by naveen521kk** [comment]
>  I did poetry add imageio to add that library. Could you please let me know if I did this correctly? I don't know how to read/interpret the poetry files

You should have marked it as dev deps and not make everybody install it. Just manully moving it pyproject.toml would do and next `poetry lock`.

**Comment by behackl** [comment]
Looking at `poetry.lock`, it seems that we already have the `imagesize` module (https://github.com/shibukawa/imagesize_py) as a dependency. In this case, I think we can avoid adding another dependency by using `imagesize.get`.

**Comment by naveen521kk** [comment]
Can this be done with Pillow?
```py
from pillow import Image
import numpy as np

im = Image.open(filename)
array = np.asarray(im)
assert array.shape == (100, 200, 4)
```

**Comment by leotrs** [comment]
Thanks all. I looked into it and it appears matplotlib can load PNGs! Silly me. Also, we need matplotlib as dev dependency anyway since it's necessary for `pytest --show_diff`

@naveen521kk I used `poetry uninstall imageio` and then `poetry add matplotlib --dev`.

**Comment by naveen521kk** [comment]
Why do you need to add matplotlib? Instead, Pillow would do, right? Or I'm wrong?

**Comment by leotrs** [comment]
Yes, we could also do this using Pillow. However, matplotlib *needs* to be added as a dev dependency anyway because it is necessary for other unrelated purposes (e.g. `pytest --show_diff` requires it). So we might as well use matplotlib for two things at once.

**Review by naveen521kk — APPROVED** [review]
LGTM!

File: pyproject.toml [hunk 1]
```diff
@@ -47,6 +47,7 @@ pytest = "^6.0"
 pylint = "*"
 guzzle_sphinx_theme = "*"
 recommonmark = "*"
+matplotlib = "^3.3.2"
```
Comment (kolibril13) [inline pyproject.toml:50]: Do we really want to add a new dependency?
I like the matplotlib library, however, manim already uses PIL for dealing with images, so why not use 
https://pillow.readthedocs.io/en/stable/reference/Image.html#functions  `PIL.Image.open` for reading the image?

**Comment by naveen521kk** [comment]
> However, matplotlib needs to be added as a dev dependency anyway because it is necessary for other unrelated purposes (e.g. pytest --show_diff requires it). 

I don't use `pytest --show_diff ` so I don't need an extra dependency just to run a test. We can use Pillow which is already included as a dependency here.

File: pyproject.toml (hunk 1 above)
Comment (kolibril13) [reply pyproject.toml:50]: Ah, sorry, I did not read the whole thread, thanks for pointing out that matplotlib needs to be added eighter way

**Comment by leotrs** [comment]
@naveen521kk Once again you have convinced me. I am using pillow now for this test, but also adding matplotlib because it is actually a dev dependency.

**Review by behackl — APPROVED** [review]
LGTM.

**Pull request merged**